File: crms/CRM.Domain/Entities/Ticket.cs
using CRM.Domain.Enums;

public class Ticket
{
    public Guid      Id              { get; set; }
    public string    Title           { get; set; } = string.Empty;
    public string    Description     { get; set; } = string.Empty;
    public string    Type            { get; set; } = "Concern";   // Concern | Inquiry | Request
    public TicketStatus Status       { get; set; }                 // Available, Claimed, Resolved, Cancelled
    public string?   ImageUrl        { get; set; }
    public Guid?     AssignedAgentId { get; set; }
    public Guid      CustomerId      { get; set; }
    public Guid?     ConversationId  { get; set; }
    public DateTime  CreatedAt       { get; set; }
    public DateTime? UpdatedAt       { get; set; }
}

File: web-crms/types/ticket.ts
export type TicketStatus =
  | "available"
  | "claimed"
  | "resolved"
  | "cancelled";

export type TicketPriority = "low" | "medium" | "high" | "urgent";
export type TicketCategory = "concern" | "inquiry" | "request" | "other";

export interface TicketCustomer {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  profileImage?: string;
}

export interface TicketAgent {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
}

export interface Ticket {
  id: string;
  title: string;
  description: string;
  status: TicketStatus;
  priority: TicketPriority;
  category: TicketCategory;
  imageUrl?: string;
  customer: TicketCustomer;
  agent?: TicketAgent;
  conversationId?: string;
  createdAt: string;
  updatedAt: string;
  resolvedAt?: string;
}

export interface PaginatedTickets {
  data: Ticket[];
  total: number;
  page: number;
  pageSize: number;
}
